docs/extensibility-http.md

```diff
@@ -0,0 +1,483 @@
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                protocol: # {{ProtocolHandler}}
+                  type: object
+                  properties:
+                    http: # {{HTTPDiscoveryHandler}}
+                      type: object
+                      properties:
+                        discoveryEndpoint:
+                          type: string
+...
+                  oneOf:
+                    - required: ["http"]
+```
+
+## Shared Configuration
+
+Revise `./shared/src/akri/configuration.rs`:
+
+```rust
+pub enum ProtocolHandler {
+    http(HTTPDiscoveryHandlerConfig),
+    ...
+}
+```
+
+And:
+
+```rust
+/// This defines the HTTP data stored in the Configuration
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct HTTPDiscoveryHandlerConfig {
+    pub discovery_endpoint: String,
+}
+```
+
+## Docker
+
+Revised `./dockerignore` to ensure `docker build ...` succeed:
+
+```console
+# Don't ignore these
+!samples/brokers/http
+!shared/
+```
+
+## Revise workspace Cargo file
+
+Revise `./Cargo.toml`:
+
+```TOML
+[workspace]
+members = [
+    ...
+    "samples/brokers/http",
+    ...
+]
+```
+
+> **NOTE ** This also ensures `http` inclusion using [rust-analyzer](https://github.com/rust-analyzer/rust-analyzer)
+
+## Revise workspace Cross file
+
+Revise `./Cross.toml`:
+
+```TOML
+[target.x86_64-unknown-linux-gnu]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:x86_64-unknown-linux-gnu-0.1.16-0.0.6"
+
+[target.arm-unknown-linux-gnueabihf]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:arm-unknown-linux-gnueabihf-0.1.16-0.0.6"
+
+[target.aarch64-unknown-linux-gnu]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:aarch64-unknown-linux-gnu-0.1.16-0.0.6"
+```
+
+## Build Akri Agent|Controller
+
+```bash
+USER=[[GTHUB-USER]]
+
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make rust-crossbuild
+
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-agent
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-controller
+```
+
+## Build|Push Broker (HTTP)
+
+```bash
+HOST="ghcr.io"
+USER=[[GITHUB-USER]]
+REPO="akri-http-broker"
+TAG="..."
+
+docker build \
+--tag=${HOST}/${USER}/${REPO}:${TAG} \
+--file=./samples/brokers/http/Dockerfile \
+. && \
+docker push ${HOST}/${USER}/${REPO}:${TAG}
+```
+
+Revise `./samples/brokers/http/http.yaml` to reflect `${IMAGE}:${TAG}`
+
+> **NOTE** If you're using a non-public repo, you can create an `imagePullSecret` to authenticate
+
+## Confirm GitHub Packages
+
+You may confirm that the `agent`, `controller` and `http` images were push to GitHub Container Registry by browsing:
+
+https://github.com/[[GITHUB-USER]]?tab=packages
+
+
+## Deploy Device(s) & Discovery
+
+```bash
+git clone https://github.com/DazWilkin/akri-http
```

Comment:
`git clone -b http-extensibility https://github.com/deislabs/akri`

Reply:
Fixed: https://github.com/deislabs/akri/pull/135/commits/af29cdba9c085dc4efa818d7880317b58b227fe4

Reply:
i'm not seeing the fix in the PR for some reason?

Reply:
strange, i can't see your latest commits in the PR's Files Changed tab ... only by clicking through in the Commits tab.

Reply:
I undid the change here: https://github.com/deislabs/akri/pull/135/commits/481e37589514a69cd3874331dcf3e7b087777fa6

`DazWilkin/akri-http` is a distinct repo that holds the Devices|Discovery examples. Per the other comment thread, I'll merge that code into this repo's `http-extensibility` branch